exporters/stats/signalfx/src/main/java/io/opencensus/exporter/stats/signalfx/SignalFxSessionAdaptor.java

```diff
@@ -0,0 +1,169 @@
+ * limitations under the License.
+ */
+
+package io.opencensus.exporter.stats.signalfx;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.signalfx.metrics.protobuf.SignalFxProtocolBuffers.DataPoint;
+import com.signalfx.metrics.protobuf.SignalFxProtocolBuffers.Datum;
+import com.signalfx.metrics.protobuf.SignalFxProtocolBuffers.Dimension;
+import com.signalfx.metrics.protobuf.SignalFxProtocolBuffers.MetricType;
+import io.opencensus.common.Function;
+import io.opencensus.common.Functions;
+import io.opencensus.stats.Aggregation;
+import io.opencensus.stats.AggregationData;
+import io.opencensus.stats.AggregationData.CountData;
+import io.opencensus.stats.AggregationData.DistributionData;
+import io.opencensus.stats.AggregationData.MeanData;
+import io.opencensus.stats.AggregationData.SumDataDouble;
+import io.opencensus.stats.AggregationData.SumDataLong;
+import io.opencensus.stats.View;
+import io.opencensus.stats.View.AggregationWindow;
+import io.opencensus.stats.ViewData;
+import io.opencensus.tags.TagKey;
+import io.opencensus.tags.TagValue;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/** Adapter for a {@code ViewData}'s contents into SignalFx datapoints. */
+final class SignalFxSessionAdaptor {
+
+  private SignalFxSessionAdaptor() {}
+
+  /**
+   * Converts the given view data into datapoints that can be sent to SignalFx.
+   *
+   * <p>The view name is used as the metric name, and the aggregation type and aggregation window
+   * type determine the metric type.
+   *
+   * @param data The {@link ViewData} containing the aggregation data of each combination of tag
+   *     values.
+   * @return A list of datapoints for the corresponding metric timeseries of this view's metric.
+   */
+  static List<DataPoint> adapt(ViewData data) {
+    View view = data.getView();
+    List<TagKey> keys = view.getColumns();
+
+    MetricType metricType = getMetricTypeForAggregation(view.getAggregation(), view.getWindow());
+    if (metricType == null) {
+      return Collections.emptyList();
+    }
+
+    List<DataPoint> datapoints = new ArrayList<>(data.getAggregationMap().size());
+    for (Map.Entry<List<TagValue>, AggregationData> entry : data.getAggregationMap().entrySet()) {
+      datapoints.add(
+          DataPoint.newBuilder()
+              .setMetric(view.getName().asString())
+              .setMetricType(metricType)
+              .addAllDimensions(createDimensions(keys, entry.getKey()))
+              .setValue(createDatum(entry.getValue()))
+              .build());
+    }
+    return datapoints;
+  }
+
+  @VisibleForTesting
+  @Nullable
+  static MetricType getMetricTypeForAggregation(Aggregation aggregation, AggregationWindow window) {
```

Comment:
I think we have the following windows:
* AggregationWindow.Interval - supports things like last minute, last hour. this is useful for in-process debugging usually, not good to export.
* AggregationWindow.Cumulative - this is the total since the registration of the view.
* AggregationWindow.Delta - we will support this in the future and will be the total since the last export.

In case of a AggregationWindow.Cumulative I think (unless you support sending cumulative data) all our aggregation can be sent as GAUGEs.

What I think you have in CUMULATIVE_COUNTER is actually what we call AggregationWindow.Delta (correct me if I am wrong).

I am not able to fully understand your COUNTER metric. Do you send deltas or total?

Reply:
Cumulative counters send monotonically increasing values, for which the absolute value doesn't matter; similar to SNMP counters. We can then calculate a time-adjusted delta to get a rate per second. Cumulative counters are allowed to rollover (for example on service restart, or overflow).

For "pure" counters, only deltas get sent. This type of counter is usually less tolerant of loss: if a datapoint doesn't make it, you loose that increment. With cumulative counters, the next value will set you back straight. For this reason, "pure" counters are usually only used for distributed counting (multiple emitters contributing increments to the same counter timeseries).

Sounds like I don't want to deal with Interval, but I should be supporting Delta. I'll fix that.

Reply:
@bogdandrutu It doesn't look like `AggregationWindow.Delta` exists yet.

Reply:
@mpetazzoni as I said AggregationWindow.Delta will be supported in the future version. I will let you know when available.

Reply:
@mpetazzoni I am waiting for you to fix the types based on the discussion.